steam_community_market/market.py

```diff
@@ -1,84 +1,66 @@
     URI = "http://steamcommunity.com/market/priceoverview"
 
     def __init__(
-        self, currency: int | str | ESteamCurrency = ESteamCurrency.USD
+        self,
+        currency: Union[
+            int, str, SteamCurrency, SteamLegacyCurrency
+        ] = SteamCurrency.USD,
     ) -> None:
-        """Sets the currency to be outputted.
+        """Initializes the Market object with a specified currency.
 
         :param currency: Currency used for prices.
-        :type currency: :class:`int`, :class:`str`, :class:`ESteamCurrency`
-        :raises SteamUnsupportedCurrency: Raised when a currency is not supported by the Steam Community Market anymore.
+        :type currency: int or str or SteamCurrency or SteamLegacyCurrency
+        :value currency: SteamCurrency.USD
+        :raises InvalidCurrencyException: Raised when a currency is considered to be invalid by the Steam Community Market.
+        :raises LegacyCurrencyException: Raised when a currency is not supported by the Steam Community Market anymore.
         """
-        if isinstance(currency, ESteamUnsupportedCurrency):
-            raise SteamUnsupportedCurrency(currency)
 
         self.currency = self._supported_currency(currency)
 
-    @staticmethod
-    def _supported_currency(
-        currency: int | str | ESteamCurrency,
-    ) -> ESteamSupportedCurrency:
-        """Returns a supported currency.
-
-        :param currency: Currency used for prices.
-        :type currency: :class:`int`, :class:`str`, :class:`ESteamCurrency`
-        :return: A supported currency.
-        :rtype: :class:`ESteamSupportedCurrency`
-
-        .. versionadded:: 1.2.4
-        """
-
-        if isinstance(currency, int):
-            return ESteamSupportedCurrency(currency)
-        if isinstance(currency, str):
-            return ESteamSupportedCurrency[currency.upper()]
-        if isinstance(currency, ESteamSupportedCurrency):
-            return currency
+    def get_overview(
+        self, name: str, app_id: Union[int, AppID]
+    ) -> Optional[dict[str, Union[bool, str]]]:
+        """Gets the prices and volume of an item in the Steam Community Market.
 
-        return ESteamSupportedCurrency.USD
-
-    def get_overview(self, name: str, app_id: int | AppID) -> dict | None:
-        """Gets the prices and volume of an item.
+        .. versionadded:: 1.0.0
+        .. versionchanged:: 1.3.0
+            * Moved :param:`app_id`'s validation to :meth:`_app_id_value`.
+            * Improved type hinting for both parameters and return value.
+            * Removed :param:`name`'s validation check and just called :meth:`_fix_item_name`.
+            * Simplified the return statement.
 
         :param name: The name of the item how it appears on the Steam Community Market.
-        :type name: :class:`str`
+        :type name: str
         :param app_id: The AppID of the game the item is from.
-        :type app_id: :class:`int`, :class:`AppID`
+        :type app_id: int or AppID

```

Comment:
```suggestion
        :type app_id: int | AppID
```
Use the standard syntax for this, I'm also not sure why you have removed the class role from these because that will potentially break inter-sphinx and make the documentation worse

Reply:
According to this, the syntax should be fine: https://www.sphinx-doc.org/en/master/tutorial/describing-code.html#cross-referencing-python-objects

Also, yes.. The `:class:` directive should be re-added for classes defined inside this library, my bad.